add differental backup with a diabled bitmap test

xutian opened the pull request:
detials steps see case RHEL-149271

Signed-off-by: Xu Tian <xutian@redhat.com>

xutian commented:
deps on https://github.com/avocado-framework/avocado-vt/pull/2127

xutian commented:
@aliang123 @ngu-niny  @CongLi  Would you like to help review this pull-request, thanks.

xutian commented:
root$ avocado run --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg 


JOB ID     : f1a210f4d318cca0f42e1a2ed92f1f0146782626
JOB LOG    : /root/avocado/job-results/job-2019-06-27T10.50-f1a210f/job.log

 (1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (76.29 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 78.63 s
JOB HTML   : /root/avocado/job-results/job-2019-06-27T10.50-f1a210f/results.html

xutian commented:
root$ avocado run --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg 
JOB ID     : fd8585520e5dd32ae650fb74fb21021dc4c51830
JOB LOG    : /root/avocado/job-results/job-2019-06-28T13.18-fd85855/job.log
 (1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (78.38 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 80.75 s
JOB HTML   : /root/avocado/job-results/job-2019-06-28T13.18-fd85855/results.html
[~]

aliang123 commented on qemu/tests/cfg/differential_backup.cfg:
```diff
@@ -0,0 +1,27 @@
+- differential_backup:
+    virt_test_type = qemu
+    type = differential_backup
+    only Linux
```
Why set "only linux" here? In fact, differential_backup won't limited to linux, windows is also supported.

xutian commented:
root$ avocado run --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg
JOB ID     : 6d5f9d99a9dab958c84b3a60b1d011922258b107
JOB LOG    : /root/avocado/job-results/job-2019-07-10T14.44-6d5f9d9/job.log
 (1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (93.54 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 96.04 s
JOB HTML   : /root/avocado/job-results/job-2019-07-10T14.44-6d5f9d9/results.html

aliang123 commented:
I run the test on RHEL8, it has the following issue, please check...
ERROR 1-Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup -> TypeError: Object of type 'map' is not JSON serializable

aliang123 commented:
Test Error with info:
(1/1) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: ERROR: %d format: a number is required, not str

aliang123 commented:
(1/1) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: FAIL: sha256 of bitmap4 not equal sha256 of bitmap1 (66.10 s)

Root cause:
  In the code: {"execute": "block-dirty-bitmap-add", "arguments": {"node": "drive_image2", "name": "bitmap_4"}, "id": "PNfR8nR6"}
 But in case: {"type": "block-dirty-bitmap-add", "data": { "node": "disk1", "name": "bitmap4","x-disabled": true} }

xutian commented:
@aliang123 

Updated, root cause is "disabled" option not pass for block_dirty_bitmap_add function in block_dirty_bitmap module. fixed has included in this pullreq.
Thanks your comments.

aliang123 commented:
Works ok with driveformat "virtio_blk" and "virtio_scsi".
(1/1) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: PASS (77.20 s)
(1/1) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (195.50 s)
(1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (200.05 s)

For driveformat=ide:
 ERROR: Shell command failed: 'parted -s -a optimal /dev/hdb mklabel gpt -- mkpart primary ext4 0% 90%'    (status: 1,    output: 'Error: Could not stat device /dev/hdb - No such file or directory.\n') (131.83 s)

aliang123 commented:
(1/1) Host_RHEL.m8.u0.qcow2.ide.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: ERROR: Shell command failed: 'parted -s -a optimal /dev/DISK mklabel gpt -- mkpart primary ext4 0% 90%'    (status: 1,    output: 'Error: Could not stat device /dev/DISK - No such file or directory.\n') (113.63 s)

xutian commented:
thanks, your testing. fixed issue and updated pullreq.

On Fri, Jul 19, 2019 at 1:32 PM aliang123 <notifications@github.com> wrote:

> (1/1)
> Host_RHEL.m8.u0.qcow2.ide.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup:
> ERROR: Shell command failed: 'parted -s -a optimal /dev/DISK mklabel gpt --
> mkpart primary ext4 0% 90%' (status: 1, output: 'Error: Could not stat
> device /dev/DISK - No such file or directory.\n') (113.63 s)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/autotest/tp-qemu/pull/1804?email_source=notifications&email_token=AAOI65FY2AXG6TM37RHSZD3QAFGVBA5CNFSM4H3XRA22YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2KTVVA#issuecomment-513096404>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOI65DXAU5V2ROD3KAV37LQAFGVBANCNFSM4H3XRA2Q>
> .
>


-- 
Regards,
Xu

aliang123 commented:
(1/1) Host_RHEL.m8.u0.qcow2.ide.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: PASS (186.77 s)
(1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.differential_backup: PASS (196.10 s)
(1/1) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: PASS (80.28 s)

aliang123 commented:
Acked-by: Aihua Liang <aliang@redhat.com>

xutian commented:
According @ngu-niny 's comments remove step to disable bitmap5

ngu-niny commented:
Test PASS:

'1-Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.differential_backup' Finshed 2019-07-22 01:42:44 2019-07-22 01:44:30 PASS 106.082321
'2-Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.differential_backup' Finshed 2019-07-22 01:44:30 2019-07-22 01:46:48 PASS 138.32221

Acked-by: Nini Gu <ngu@redhat.com>

aliang123 commented:
(1/1) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.differential_backup: PASS (76.99 s)

Acked.

xutian commented:
Ignore CI check results, because the failure caused by CI environment.

aliang123 commented:
id: 1824053
